test/semi_hdp.cc

```diff
@@ -1,5 +1,6 @@
 #include <gtest/gtest.h>
```

Comment:
This file now fails 3 tests :( I have no idea why. Here's the log:
```
[----------] 5 tests from semihdp
[ RUN      ] semihdp.relabel
[       OK ] semihdp.relabel (2 ms)
[ RUN      ] semihdp.relabel2
[       OK ] semihdp.relabel2 (2 ms)
[ RUN      ] semihdp.sample_unique_values
/media/bruno/bbayesmix/test/semi_hdp.cc:207: Failure
Expected: (std::dynamic_pointer_cast<NNIGHierarchy>(sampler.get_table(0, 0)) ->get_state() .mean)
 > (0), actual: -0.747293 vs 0
[  FAILED  ] semihdp.sample_unique_values (3 ms)
[ RUN      ] semihdp.sample_unique_values2
/media/bruno/bbayesmix/test/semi_hdp.cc:289: Failure
Expected: (std::dynamic_pointer_cast<NNIGHierarchy>(sampler.get_shared_table(0)) ->get_state()
 .mean) > (0), actual: -5.52852 vs 0
[  FAILED  ] semihdp.sample_unique_values2 (3 ms)
[ RUN      ] semihdp.sample_allocations1
/media/bruno/bbayesmix/test/semi_hdp.cc:352: Failure
Expected equality of these values:
  snew[0][1]
    Which is: 10
  0
[  FAILED  ] semihdp.sample_allocations1 (19 ms)
[----------] 5 tests from semihdp (29 ms total)
```
They work fine on `master`. Git history confirms this file and the SemiHDP sampler ones haven't been touched for ages, except for the recent renaming of AlgorithmState, which obviously has nothing to do with it.

Reply:
This is likely due to some changes at BaseHierarchy/ConjugateHierarchy. Note that the SemiHDP calls the overloaded sample_full_cond (the version that passes observations to hierarchies)

Reply:
Do you know why it broke? Because the default argument value in `sample_full_cond(update_params)` is `true`, not `false` like I said [before](https://github.com/bayesmix-dev/bayesmix/pull/57#issuecomment-787097303). Not the proudest hours of debugging of my life...

Reply:
There is a nice lesson to be learnt here :)